src/esportsbot/cogs/MusicCog.py

```diff
@@ -0,0 +1,1117 @@
+        queue_message_instance: Message = await music_channel_instance.fetch_message(queue_message_id)
+        preview_message_instance: Message = await music_channel_instance.fetch_message(preview_message_id)
+
+        # Update the messages
+        await queue_message_instance.edit(content=queue_message)
+        await preview_message_instance.edit(embed=preview_message)
+
+    def __make_updated_queue_message(self, guild_id):
+        """
+        Update the queue message in a given guild.
+        :param guild_id: The guild id of the guild to update the queue message in.
+        :return: A string of the queue that is to be the new queue message.
+        """
+
+        if not self._currently_active.get(guild_id) or len(self._currently_active.get(guild_id).get('queue')) == 0:
+            # If the queue is empty or the bot isn't active
+            updated_queue_message = self._empty_queue_message
+        else:
+            updated_queue_message = self.__make_queue_list(guild_id)
+
+        return updated_queue_message
+
+    def __make_update_preview_message(self, guild_id):
+        """
+        Update the preview message in a given guild.
+        :param guild_id: The guild id of the guild to update the preview message in.
+        :return: An embed message for the updated preview message in a given guild id.
+        """
+
+        if not self._currently_active.get(guild_id) or len(self._currently_active.get(guild_id).get('queue')) == 0:
+            # If the queue is empty, provide the empty queue embed or the bot isn't active
+            updated_preview_message = self._no_current_song_message
+        else:
+            current_song = self._currently_active.get(guild_id).get('current_song')
+            updated_preview_message = Embed(title="Currently Playing: " + current_song.get('title'),
+                                            colour=Colour(0xd462fd), url=current_song.get('link'),
+                                            video=current_song.get('link'))
+            updated_preview_message.set_image(url=current_song.get('thumbnail'))
+
+        self.__add_time_remaining_field(guild_id, updated_preview_message)
+
+        return updated_preview_message
+
+    def __generate_link_data_from_queue(self, guild_id):
+        """
+        Get the stream data, thumbnails and other data regarding the top song in the queue.
+        :param guild_id: The guild id to get the queue from.
+        :return: A dict of information required to play a song.
+        """
+        current_song = self._currently_active.get(guild_id).get('queue')[0]
+        return self.__format_download_data(self.__download_video_info(current_song.get('link')))
+
+    def __download_link_list_info(self, links):
+        """
+        Downloads the basic information regarding a list of links.
+        :param links: The list of video links.
+        :return: A dict of each link with their video titles.
+        """
+        threads = min(20, len(links))
```

Comment:
What if we don't have 20 CPU threads?
By default, `ThreadPoolExecutor` will use the number of threads available on the system, so I personally wouldn't pass `max_workers` at all